### PR TITLE
formatting and parsing messages

### PR DESCRIPTION
**Pull request by caiodasilva2005**

### CODE REVIEW

File: .vscode/settings.json
Comment (vladaviedov) [inline .vscode/settings.json:1]: gitignore this folder

File: shared/include/messages.hpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#ifndef MESSAGES_HPP
+#define MESSAGES_HPP
+
+#include <cstdint>
+#include <string>
+
+struct msg_t
+{
```
Comment (vladaviedov) [inline shared/include/messages.hpp]: Put bracket on the same line

File: shared/include/messages.hpp [hunk 2]
```diff
@@ -0,0 +1,18 @@
+#ifndef MESSAGES_HPP
```
Comment (vladaviedov) [inline shared/include/messages.hpp]: use #pragma once

File: shared/include/messages.hpp [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+struct msg_t
+{
+    std::string caller_id;
+    std::string receiver_id;
+    std::string body;
+};
+
+
+std::string format_message(const msg_t &msg);
+msg_t parse_message(std::string str_msg);
```
Comment (vladaviedov) [inline shared/include/messages.hpp]: Pass by reference

File: shared/src/messages.cpp [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+std::string format_message(const msg_t &msg) {
+    std::string str_msg("AIRv1.0 " + msg.receiver_id + " SM\n" + msg.body + "\nEM " + msg.caller_id);
+    return str_msg;
+}
+
+msg_t parse_message(std::string str_msg) {
+    const msg_t msg = {
+        .caller_id = str_msg.substr(8,9),
+        .receiver_id = str_msg.substr(20,21),
```
Comment (vladaviedov) [inline shared/src/messages.cpp]: Message body is dynamically sized, you can't do this

File: shared/include/messages.hpp [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+#endif
```
Comment (vladaviedov) [inline shared/include/messages.hpp]: Files should end with an empty line

**Review by vladaviedov — CHANGES_REQUESTED** [review]
The body of the message does not have a fixed length.

File: shared/src/messages.cpp [hunk 6]
```diff
@@ -1,18 +1,38 @@
...
-        .receiver_id = str_msg.substr(str_msg.find("EM") + 2, str_msg.length()),
-        .body = str_msg.substr(str_msg.find("SM") + 2, str_msg.find(" EM"))
-    };
-    return msg;
-}
+msg_t parse_message(const std::string &str_msg) {
 
+    size_t start_msg = str_msg.find("SM");
+    size_t end_msg = str_msg.find("EM");
+    std::string msg_header = "AIRv1.0";
```
Comment (vladaviedov) [inline shared/src/messages.cpp]: mark as constexpr

**Comment by caiodasilva2005** [comment]
made a new branch because I did something stupid in this one

**Pull request closed without merging**